mis_builder/static/src/components/mis_report_widget.js

```diff
@@ -0,0 +1,117 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+
+const { Component, onWillStart } = owl;
+
+export class MisReportWidget extends Component {
+    setup() {
+        super.setup();
+        this.orm = useService('orm');
+        this.user = useService("user");
+        this.legacyActionManager = useService("legacy_action_manager");
+        this.JSON = JSON;
+        this.user_context = Component.env.session.user_context;
+
+        onWillStart(this.willStart);
+    }
+    // Lifecycle
+    async willStart() {
+        this.props.mis_report_data = await this.orm.call(
+                "mis.report.instance",
+                "compute",
+                [this._instanceId()],
+                {context: this.user_context}
+        );
+
+        this.props.show_settings = await this.user.hasGroup("account.group_account_user");
+
+
+        this.props.has_group_analytic_accounting = await this.user.hasGroup("analytic.group_analytic_accounting");
+
+        var result = await this.orm.call(
+                "mis.report.instance",
+                "read",
+                [this._instanceId(), ["hide_analytic_filters"]],
+                {context: this.user_context},
+            )
+        this.props.hide_analytic_filters = result && result[0].hide_analytic_filters;
+
+    }
+    /**
+     * Return the id of the mis.report.instance to which the widget is
+     * bound.
+     *
+     * @returns int
+     */
+    _instanceId () {
+        if (this.props.value) {
+            return this.props.value;
+        }
+
+        /*
+            * This trick is needed because in a dashboard the view does
+            * not seem to be bound to an instance: it seems to be a limitation
+            * of Odoo dashboards that are not designed to contain forms but
+            * rather tree views or charts.
+            */
+        var context = this.props.record.context;
+        if (context.active_model === "mis.report.instance") {
+            return context.active_id;
+        }
+    }
+    async drilldown (event) {
+        const drilldown = $(event.target).data("drilldown");
+        const action = await this.orm.call(
+            "mis.report.instance",
+            "drilldown",
+            [this._instanceId(), drilldown],
+            {context: this.user_context},
+        )
+        this.legacyActionManager.do_action(action);
```

Comment:
Do you know why we need to use a "legacy" action manager here? Sorry for the maybe dumb question, I'm not very familiar with the frontend framework.

Reply:
Thanks for the question, it basically makes the legacy `do_action` work with the owl action service, you can see [this](https://github.com/odoo/odoo/blob/16.0/addons/web/static/src/legacy/backend_utils.js#L254) for references

This allows us to click on the report numbers for drilldown-ing the details

![image](https://user-images.githubusercontent.com/46039081/219312158-f351a8ba-cfc6-4faa-b227-ad1ee5b41cb7.png)

Reply:
anw because of your nice question, we can use this `this.action = useService("action");` instead indeed

Reply:
Great! Thanks for investigating.